File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

	long int square_of_sum(int n)
	{
		long int sum = (n * (n + 1)) / 2;
		return sum * sum;

	}

	long int sum_of_squares(int n)
	{
		long int sum = ((2 * n + 1) * n * (n + 1)) / 6;
		return sum;
	}

	long int difference(int n)
	{
		return square_of_sum(n) - sum_of_squares(n);
	}

}  // namespace difference_of_squares

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(number) {
    this.number = number
  }

  get sumOfSquares() {
    const n = this.number;
    const sum = ((2 * n + 1) * n * (n + 1)) / 6;
    Object.defineProperty(this, 'sumOfSquares', { value: sum });
    return sum;
  }

  get squareOfSum() {
    const n = this.number
    const sum = (n * (n + 1)) / 2;
    const sumSq = sum * sum;
    Object.defineProperty(this, 'squareOfSum', { value: sumSq });
    return sumSq;
  }

  get difference() {
    const diff = this.squareOfSum - this.sumOfSquares;
    Object.defineProperty(this, 'difference', { value: diff });
    return diff;
  }
}
